src/climate_learn/loaders.py

```diff
@@ -0,0 +1,356 @@
+        else:
+            raise_not_impl()
+    return model, optimizer, lr_scheduler
+
+def load_optimizer(
+    net: torch.nn.Module,
+    optim: str,
+    optim_kwargs: Dict[str, Any] = {}
+):
+    if len(list(net.parameters())) == 0:
+        warnings.warn(
+            "Net has no trainable parameters, setting optimizer to `None`"
+        )
+        optimizer = None
+    if optim.lower() == "sgd":
+        optimizer = torch.optim.SGD(net.parameters(), **optim_kwargs)
+    elif optim.lower() == "adam":
+        optimizer = torch.optim.Adam(net.parameters(), **optim_kwargs)
+    elif optim.lower() == "adamw":
+        optimizer = torch.optim.AdamW(net.parameters(), **optim_kwargs)
+    else:
+        raise NotImplementedError(
+            f"{optim} is not an implemented optimizer. If you think it should"
+            " be, please raise an issue at"
+            " https://github.com/aditya-grover/climate-learn/issues"
+        )
+    return optimizer
+
+def load_lr_scheduler(
+    sched: str,
+    optimizer: torch.optim.Optimizer,
+    sched_kwargs: Dict[str, Any] = {}
+):
+    if optimizer is None:
+        warnings.warn(
+            "Optimizer is `None`, setting LR scheduler to `None` too"
+        )
+        lr_scheduler = None
+    if sched == "constant":
+        lr_scheduler = LRScheduler.ConstantLR(optimizer, **sched_kwargs)
+    elif sched == "linear":
+        lr_scheduler = LRScheduler.LinearLR(optimizer, **sched_kwargs)
+    elif sched == "exponential":
+        lr_scheduler = LRScheduler.ExponentialLR(optimizer, **sched_kwargs)
+    elif sched == "linear-warmup-cosine-annealing":
+        lr_scheduler = LinearWarmupCosineAnnealingLR(optimizer, **sched_kwargs)    
+    else:
+        raise NotImplementedError(
+            f"{sched} is not an implemented learning rate scheduler. If you"
+            " think it should be, please raise an issue at"
+            " https://github.com/aditya-grover/climate-learn/issues"
+        )
+    return lr_scheduler
+
+def load_loss(loss_name, metainfo):
+    loss_cls = METRICS_REGISTRY.get(loss_name, None)
+    if loss_cls is None:
+        raise NotImplementedError(
+            f"{loss_name} is not an implemented loss. If you think it should be,"
+            " please raise an issue at"
+            " https://gtihub.com/aditya-grover/climate-learn/issues"
+        )
+    loss = loss_cls(
+        aggregate_only=True,
+        metainfo=metainfo
+    )
+    return loss
+
+def load_transform(transform_name, data_module):
+    transform_cls = TRANSFORMS_REGISTRY.get(transform_name, None)
+    if transform_cls is None:
+        raise NotImplementedError(
+            f"{transform_name} is not an implemented transform. If you think"
+            " it should be, please raise an issue at"
+            " https://github.com/aditya-grover/climate-learn/issues"
+        )
+    transform = transform_cls(data_module)
+    return transform
+        
+def get_data_dims(data_module):
```

Comment:
Can you please update the logic for `get_data_dims`, such that we don't make a call to `train_dataloader()`. My reason for asking so, is because of the performance reasons for `ShardDataset` i.e. we would need to load a chunk into memory.

I believe, in the current PR we can assume that the `.setup()` has been called for the `train_dataset()`. (**please also add this assumption as a comment, so that we can later revisit this issue and remove this assumption and add corresponding error handling code**). If we make this assumption, then we can use `Task` and logic inside `collate_fn() ` to return the **input** (`x`) and **output** (`y`) shapes.

Reply:
I don't understand what you are asking me to do. `collate_fn` requires a batch anyway, so wouldn't that also require me to use the dataloader?

Reply:
Let's ignore this then for now. I will open a new PR for this once this gets merged.